adr/ADR-0003-trustyai-service-deployment-using-operator-pattern.md

```diff
@@ -0,0 +1,245 @@
+
+The processor configuration is embedded in a Kubernetes ConfigMap and follows the format:
+
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: model-serving-config
+  namespace: default
+data:
+  config.yaml: |
+    payloadProcessors: http://trustyai-service.$NAMESPACE/consumer/kserve/v2
+```
+
+In this configuration, `$NAMESPACE` is replaced by the Operator with the namespace where the TrustyAI Service and ModelMesh Serving are deployed ensuring that ModelMesh sends payloads correctly to the TrustyAI Service.
+
+### Monitoring (Prometheus)
+
+The TrustyAI Operator also creates a `ServiceMonitor` object which defines the services to be monitored by Prometheus. The `ServiceMonitor` will have the following configuration:
+
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: trustyai-metrics
+  labels:
+    modelmesh-service: modelmesh-serving
+spec:
+  endpoints:
+    - interval: 4s
+      path: /q/metrics
+      honorLabels: true
+      honorTimestamps: true
+      scrapeTimeout: 3s
+      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      targetPort: 8080
+      scheme: http
+      params:
+   	 'match[]':
+ 		 - '{__name__= "trustyai_spd"}'
+ 		 - '{__name__= "trustyai_dir"}'
+      metricRelabelings:
+   	 - action: keep
+ 		 regex: trustyai_.*
+ 		 sourceLabels:
+   		 - __name__
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: trustyai-service
+```
+
+
+The `ServiceMonitor` object targets the TrustyAI Service and specifies how Prometheus should scrape metrics from the service, which includes the path to the metrics endpoint (`/q/metrics`), the interval at which it should scrape the metrics (every 4 seconds), and the type of metrics it should scrape (metrics with names that start with `trustyai_`).
+The selector would also be updated to match the labels of the TrustyAI Service from the Custom Resource.
+The scrape interval and metrics names could potentially also be configurable via the custom resource (with the current values as defaults).
+
+
+A possibility for the service monitor customization is the inclusion of custom values using a nested `ref:` field for `serviceMonitoring`. This would allow users to specify a custom configuration for the `ServiceMonitor` object. For example:
+
+
+```yaml
+apiVersion: trustyai.opendatahub.io/v1
+kind: TrustyAIService
+metadata:
+  name: trustyai-service-example
+  namespace: default
+spec:
+  ...
+  serviceMonitoring:
+    ref:
+      apiVersion: monitoring.coreos.com/v1
+      kind: ServiceMonitor
+      ...
+      spec:
+   	 endpoints:
+ 		 - interval: 15s
+```
+
+If such configuration is not provided, the operator will use the default configuration.
+
+### Route
```

Comment:
Copying a comment from the google doc -- we should think through the servicemesh integration (if any) here too.